src/utils/writeHTML.js

```diff
@@ -0,0 +1,44 @@
+/**
+ * Replace the contents of `element` with the given `html` string, running all scripts contained within it.
+ *
+ * @param element the container element. Must be in the DOM.
+ * @param html HTML string
+ */
+export function replaceHTML(element, html) {
```

Comment:
My only concern with this approach is that I believe it makes scripts execute asynchronously, where as using `document.write` will execute synchronous. So you might hit some edge cases here.